app/Http/Controllers/CategoryController.php

```diff
@@ -30,4 +35,104 @@ public function store(Request $request)
             ]);
         }
     }
+
+    public function show($id)
+    {
+        $category = Category::find($id);
+        if (is_null($category)) {
+            return response()->json([
+                'status' => 404,
+                'errors' => 'No category found to be shown!',
+            ]);
+        }
+
+        return response()->json([
+            'status' => 200,
+            'category' => $category,
+        ]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $validator = Validator::make($request->all(), [
+            'name' => ['required', 'string', 'max:40', 'regex:/(^([a-zA-Z]+)(\d+)?$)/u', 'unique:categories'],
```

Comment:
Not sure if it will work with `unique:categories` In case admin didn't update the name, Please double check